SYCL/Basic/device-selectors.cpp

```diff
@@ -0,0 +1,88 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN:  %t.out
+
+#include <sycl/sycl.hpp>
+using namespace sycl;
+
+auto exception_handler_lambda = [](exception_list elist) {
+  // nop
+};
+
+void handle_exceptions_f(exception_list elist) {
+  // nop
+}
+
+auto refuse_any_device_lambda = [](const device &d) { return -1; };
+
+int refuse_any_device_f(const device &d) { return -1; }
+
+int main() {
+  // Ensure these compile correctly. That exception handler does not
+  // end up specialized against device selector.
+  queue exception_queue_01(exception_handler_lambda);
+  queue exception_queue_02(handle_exceptions_f);
+
+  // Set up callable device selector.
+  std::vector<device> deviceList = device::get_devices();
+  device lastDevice = deviceList.back();
```

Comment:
```suggestion
  device &lastDevice = deviceList.back();
```

I think. Sorry for not spotting earlier.

Reply:
no worries. got it.